centinel/experiments/baseline.py

```diff
@@ -0,0 +1,271 @@
+# every URL in the list.
+#
+# Input files can be either simple URL lists or CSV
+# files. In case of CSV input, the first column is
+# assumed to be the URL and the rest of the columns
+# are included in the results as metadata.
+
+
+import os
+import logging
+import time
+import urlparse
+
+from centinel.experiment import Experiment
+from centinel.primitives import dnslib
+from centinel.primitives.tcpdump import Tcpdump
+from centinel.primitives import tls
+import centinel.primitives.http as http
+import centinel.primitives.traceroute as traceroute
+
+
+class BaselineExperiment(Experiment):
+    name = "baseline"
+    # country-specific, world baseline
+    # this can be overridden by the main thread
+    input_files = ['country', 'world']
+
+    # we do our own tcpdump recording here
+    overrides_tcpdump = True
+
+    def __init__(self, input_files):
+        self.input_files = input_files
+        self.results = []
+
+        # should we do tcpdump?
+        if os.geteuid() != 0:
+            self.record_pcaps = False
+        else:
+            self.record_pcaps = True
+
+        if os.geteuid() != 0:
+            logging.info("Centinel is not running as root, "
+                         "traceroute will be limited to UDP.")
+            self.traceroute_methods = ["udp"]
+        else:
+            # if running as root, TCP and UDP traceroute
+            # doing all 3 methods takes a lot of time
+            # self.traceroute_methods = ["icmp", "udp", "tcp"]
+            self.traceroute_methods = ["tcp", "udp"]
+
+    def run(self):
+        if self.record_pcaps:
+            self.external_results = {}
+
+        for input_file in self.input_files.items():
+            logging.info("Testing input file %s..." % (input_file[0]))
+            self.results.append(self.run_file(input_file))
+
+    def run_file(self, input_file):
+        file_name, file_contents = input_file
+
+        # Initialize the results for this input file.
+        # This can be anything from file name to version
+        # to any useful information.
+        result = {}
+        result["file_name"] = file_name
+
+
+        http_results = {}
+        tls_results = {}
+        dns_results = {}
+        traceroute_results = {}
+        url_metadata_results = {}
+        file_metadata = {}
+        file_comments = []
+
+        # each pcap file is stored in a separate file
+        # designated by a number. the indexes are stored
+        # in the json file and the pcap files are stored
+        # with their indexes as file names.
+        pcap_results = {}
+        pcap_indexes = {}
+        url_index = 0
+        comments = ""
+
+        # we may want to make this threaded and concurrent
+        for line in file_contents:
+            line = line.strip()
+
+            # parse file comments, if it looks like "key : value",
+            # parse it as a key-value pair. otherwise, just
+            # store it as a raw comment.
+            if line[0] == '#':
```

Comment:
just make the check for a comment into a boolean variable and use that in later if statements. It's clearer with low overhead